knowledge/android/MASVS-STORAGE/MASTG-KNOW-0051.md

```diff
@@ -18,15 +18,12 @@ The latter requirement gives developers direct memory access. Make sure that the
 
 Using non-primitive mutable types like `StringBuffer` and `StringBuilder` may be acceptable, but it's indicative and requires care. Types like `StringBuffer` are used to modify content (which is what you want to do). To access such a type's value, however, you would use the `toString` method, which would create an immutable copy of the data. There are several ways to use these data types without creating an immutable copy, but they require more effort than using a primitive array. Safe memory management is one benefit of using types like `StringBuffer` , but this can be a two-edged sword. If you try to modify the content of one of these types and the copy exceeds the buffer capacity, the buffer size will automatically increase. The buffer content may be copied to a different location, leaving the old content without a reference use to overwrite it.
 
-Unfortunately, few libraries and frameworks are designed to allow sensitive data to be overwritten. For example, destroying a key, as shown below, doesn't remove the key from memory:
-
-```kotlin
-val secretKey: SecretKey = SecretKeySpec("key".toByteArray(), "AES")
-secretKey.destroy()
-```
-
-Overwriting the backing byte-array from `secretKey.getEncoded` doesn't remove the key either; the SecretKeySpec-based key returns a copy of the backing byte-array. See the sections below for the proper way to remove a `SecretKey` from memory.
-
-The RSA key pair is based on the `BigInteger` type and therefore resides in memory after its first use outside the `AndroidKeyStore`. Some ciphers (such as the AES `Cipher` in `BouncyCastle`) do not properly clean up their byte-arrays.
-
```

Comment:
I removed this paragraph, there is no destroy method for [SecretKeySpec](https://developer.android.com/reference/javax/crypto/spec/SecretKeySpec) and the sentence "See the sections below for the proper way to remove a SecretKey from memory." was redundant.

Reply:
I brought back these paragraphs but in a better shape as I think the info is relevant and provides context (added more links for additional context). See lines 21-26.